### PR TITLE
Adding if check to stop blank comments

### PR DESCRIPTION
**Pull request by adamdavies1**
You should no longer be able to add blank comments

### CODE REVIEW

**Review by stevewalton28 — APPROVED** [review]

File: app/assess/data.py [hunk 1]
```diff
@@ -467,21 +467,23 @@ def get_comments(application_id: str, sub_criteria_id: str, theme_id, themes):
...
-        "comment_type": "COMMENT",
-        "theme_id": theme_id,
-    }
-    url = Config.ASSESSMENT_COMMENT_ENDPOINT
-    response = requests.post(url, json=data_dict)
-    current_app.logger.info(
-        f"Response from Assessment Store: '{response.json()}'."
-    )
+    # TODO: should we add something that informs the user the comment doesnt hasnt saved?
+    if comment.isspace() == False and comment != "":
```
Comment (gidsg) [inline app/assess/data.py:471]: We should be using a form validator for this rather than by hand https://wtforms.readthedocs.io/en/2.3.x/validators/

**Review by gidsg — CHANGES_REQUESTED** [review]
See comment, we should be using a form validator for the input, this would produce the correct flash message.

**Comment by gidsg** [comment]
@adamdavies1 i.e. add a validator here https://github.com/communitiesuk/funding-service-design-assessment/blob/7186947638a792aef8e67337f47eb76af55d197a/app/assess/forms/comments_form.py#L8

File: app/assess/data.py (hunk 1 above)
Comment (adamdavies1) [reply app/assess/data.py]: Is now using validation

File: app/assess/forms/comments_form.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 from flask_wtf import FlaskForm
 from wtforms import TextAreaField
 from wtforms.validators import length
+from wtforms.validators import DataRequired
 
 
 class CommentsForm(FlaskForm):
 
-    comment = TextAreaField("Comment", validators=[length(max=10000)])
+    comment = TextAreaField("Comment", validators=[length(max=5), DataRequired()])
```
Comment (gidsg) [inline app/assess/forms/comments_form.py]: Looks like this is setting the length to max 5 characters?

File: app/assess/templates/macros/comments_box.html [hunk 3]
```diff
@@ -15,6 +15,9 @@
       "classes": "govuk-label--s",
       isPageHeading: true
     },
+    "errorMessage": {
+      "text": "A comment must not be empty"
+  } if commentForm.comment.errors
```
Comment (gidsg) [inline app/assess/templates/macros/comments_box.html:20]: Could be too long or empty so this should handle both cases, can we pass through the errors we get?

File: app/assess/routes.py [hunk 4]
```diff
@@ -124,7 +124,7 @@ def display_sub_criteria(
 
         comment_form = CommentsForm()
 
-        if comment_form.validate_on_submit():
+        if request.method == "POST" and comment_form.validate_on_submit():
```
Comment (harryyo) [inline app/assess/routes.py:127]: From the docs [validate_on_submit()](https://flask-wtf.readthedocs.io/en/0.15.x/quickstart/#validating-forms) also checks if the request.method is post so this shouldn't be needed I think?

File: app/assess/routes.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def display_sub_criteria(
...
```
Comment (harryyo) [inline app/assess/routes.py:175]: As above

File: app/assess/templates/macros/comments_box.html (hunk 3 above)
Comment (harryyo) [reply app/assess/templates/macros/comments_box.html:20]: Could do a catch all wording like "A comment must be between 1-1000 characters"

**Review by harryyo — CHANGES_REQUESTED** [review]
Just a couple of comments

**Review by harryyo — APPROVED** [review]
👍

**Review by gidsg — APPROVED** [review]

File: app/assess/forms/comments_form.py [hunk 6]
```diff
@@ -1,8 +1,9 @@
 from flask_wtf import FlaskForm
 from wtforms import TextAreaField
 from wtforms.validators import length
+from wtforms.validators import DataRequired
 
 
 class CommentsForm(FlaskForm):
 
-    comment = TextAreaField("Comment", validators=[length(max=10000)])
+    comment = TextAreaField("Comment", validators=[length(max=10000), DataRequired()])
```
Comment (gidsg) [inline app/assess/forms/comments_form.py:9]: If you want a more friendly message, you can pass a `message` argument to the validator.

**Pull request merged**